File: streampipes-model/src/main/java/org/streampipes/model/modelconnect/AdapterStreamDescription.java
package org.streampipes.model.modelconnect;
import org.streampipes.empire.annotations.Namespaces;
import org.streampipes.empire.annotations.RdfProperty;
import org.streampipes.empire.annotations.RdfsClass;
import org.streampipes.model.SpDataSet;
import org.streampipes.model.SpDataStream;

import javax.persistence.Entity;

@Namespaces({"sp", "https://streampipes.org/vocabulary/v1/"})
@RdfsClass("sp:AdapterStreamDescription")
@Entity
public class AdapterStreamDescription extends AdapterDescription {

    public AdapterStreamDescription() {
    }

    public AdapterStreamDescription(FormatDescription formatDescription, ProtocolDescription protocolDescription) {
        super(formatDescription, protocolDescription);
    }

    @RdfProperty("sp:hasDataStream")
    private SpDataStream dataStream;

    public SpDataStream getDataStream() {
        return dataStream;
    }

    public void setDataStream(SpDataStream dataStream) {
        this.dataStream = dataStream;
    }
}

File: src/app/connect/model/AdapterStreamDescription.ts
import {RdfProperty} from '../tsonld/RdfsProperty';
import {RdfsClass} from '../tsonld/RdfsClass';
import {DataSetDescription} from './DataSetDescription';
import {AdapterDescription} from './AdapterDescription';
import {DataStreamDescription} from './DataStreamDescription';

@RdfsClass('sp:AdapterStreamDescription')
export class AdapterStreamDescription extends AdapterDescription {

    @RdfProperty("sp:hasDataStream")
    public dataStream: DataStreamDescription;

    constructor(id: string) {
        super(id)
    }
}